apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/StorageClientImplTest.cs

```diff
@@ -69,5 +76,402 @@ public void GetEffectiveEncryptionKey_Invalid(EncryptionKey clientkey, Encryptio
             var exception = Assert.Throws<ArgumentException>(() => client.GetEffectiveEncryptionKey(operationKey, kmsKey));
             Assert.Contains(expectedMessageSubstring, exception.Message);
         }
+
+        // Tests for GCS Retry :
```

Comment:
These comments were just off-the-cuff - I'd suggest either rewriting them more thoroughly, or deleting them.
(We could potentially make this StorageClientImplRetryTest.cs as a name as well, to separate it out from any other StorageClientImpl tests we write.)